### ISSUE TITLE
How to use batch norm when there has padding?

### ISSUE THREAD

**Issue by RichardHWD** [pytorch/pytorch#25910]
I am doing a NLP task and input various-length sentences into model, so I add some padding to keep every sentence to the longest length in one batch. However, padding length can affect batch norm obviously. And I cannot find a module with a parameter of mask? Can you help me ?

**Comment by soumith** [comment]
please use https://discuss.pytorch.org for questions
